CMakeLists.txt

```diff
@@ -162,18 +162,25 @@ if(MSVC)
 		set(MSVC_DISABLED_WARNINGS_LIST
 			"C4018" # 'expression' : signed/unsigned mismatch
 			"C4057" # 'operator' : 'identifier1' indirection to
-		        	# slightly different base types from 'identifier2'
+			        # slightly different base types from 'identifier2'
+			"C4098" # 'function' : void function returning a value
 			"C4100" # 'identifier' : unreferenced formal parameter
 			"C4127" # conditional expression is constant
+			"C4132" # 'object' : const object should be initialized
 			"C4146" # unary minus operator applied to unsigned type,
 			        # result still unsigned
+			"C4206" # nonstandard extension used : translation unit is empty
 			"C4244" # 'argument' : conversion from 'type1' to 'type2',
 			        # possible loss of data
 			"C4245" # 'conversion' : conversion from 'type1' to 'type2',
 			        # signed/unsigned mismatch
 			"C4267" # 'var' : conversion from 'size_t' to 'type',
 			        # possible loss of data
+			"C4295" # 'array' : array is too small to include a terminating
+			        # null character
 			"C4389" # 'operator' : signed/unsigned mismatch
+			"C4701" # Potentially uninitialized local variable 'name' used
```

Comment:
If there are specific false positives it makes sense to just push and pop the warnings around them and I can just remove this line when rebasing.

Reply:
```
2024-11-12T11:43:53.5610637Z D:\a\portable\portable\crypto\bn\bn_convert.c(110,1): warning C4701: potentially uninitialized local variable 'w' used [D:\a\portable\portable\build\crypto\crypto_obj.vcxproj]
```
This is a false positive. `w` is initialized in the first iteration since `b == BN_BITS2`.

```
2024-11-12T11:43:59.3244563Z D:\a\portable\portable\crypto\pem\pem_lib.c(475,1): warning C4701: potentially uninitialized local variable 'i' used [D:\a\portable\portable\build\crypto\crypto_obj.vcxproj]
2024-11-12T11:43:59.3407853Z D:\a\portable\portable\crypto\pem\pem_lib.c(424,1): warning C4701: potentially uninitialized local variable 'j' used [D:\a\portable\portable\build\crypto\crypto_obj.vcxproj]
```

Likewise. The line is only reached if the variable i or j is initialized (this is terrible code...)

These are the only three that I can find, so I would prefer if we disabled these warnings on bn_convert and pem_lib only.